### ISSUE TITLE
Some user snippets don't want to appear in the snippets list

### ISSUE THREAD

**Issue by user3323** [Microsoft/vscode#32231]
- VSCode Version: 1.14.2
- OS Version: win7x64

Steps to Reproduce:

Here is my snippet for **getElementsByClassName** in **javascript.json** file:

```
"getElementsByClassName": {
  "prefix": "cla",
  "body": [
    "getElementsByClassName(\"$1\");$0"
  ],
  "description": "getElementsByClassName"
}
```
So, as you can see in GIF, VS Code doesn't want to include my snippet right after typing, for example, **document.body.**, and it appears only after removing 1 or more letters from "cla" and starting again.

Should be fixed.

GIF:

![snippets list _ 2](https://user-images.githubusercontent.com/24613274/29177592-bf112f6a-7df7-11e7-8129-1f13ff6e0e80.gif)

**Comment by jrieken** [comment]
duplicate of https://github.com/Microsoft/vscode/issues/22205

**Comment by user3323** [comment]
@jrieken omg. This problem for me is unbearable. I've read https://github.com/Microsoft/vscode/issues/22205, but I didn't really get what you wrote in the last message:
> We won't trigger snippets after trigger character because they are kind of private to the extensions that have registered for it.
Sorry, my English is not perfect, maybe that's why.

I really don't understand how everyone can deal with it. 70-80% of all my custom snippets goes after the `. `(dot). I write my own snippets because I don't like basic snippets.

Consider this:
If I need to get elements by class name from any element:

- the basic snippet system offers you to type some (2, 3 or more) letters, then you will see your snippet in the list (it can be on the 3rd, 5th or whatever place in the list) and you need to press some down arrows, and after that finally press Enter... and it will expand into "dry" method `getElementsByClassName`, without parenthesis, without quote marks (as you would expect)
- the custom snippet system offers you **everything**! You know EXACTLY what letters you need to type, how many letters (1, 2 or 3). You don't need to press Arrow buttons. Just 2-3 letters and Enter... and it becomes into method you wrote it yourself - `getElementsByClassName("$1");$0`

It's like to compare basic Windows notepad with VS Code.

And, as you can see in the GIF, snippets list will include custom snippet after removing one letter: cl --> c --> again cl --> and it will appear in the list. I really don't get this trigger algorithm.

How can we stay with basics, if it really makes coding harder for everyone?

If you don't want to implement this "showing custom snippets after dot" feature by default, can you at least make an option in the settings, so we can turn it to `true`?

Thanks in advance!

**Comment by jrieken** [comment]
> and it will expand into "dry" method getElementsByClassName, without parenthesis, without quote marks (as you would expect)

There are setting for JavaScript and TypeScript to change this (`    "typescript.useCodeSnippetsOnMethodSuggest": true` and `"javascript.useCodeSnippetsOnMethodSuggest": true`). When set method/function don't complete "dry" but with the invocation operator and has arguments pre-filled (as snippets). That setting might help you. 

![aug-16-2017 12-47-44](https://user-images.githubusercontent.com/1794099/29359985-229a2b3a-8281-11e7-9d2c-385756a13812.gif)

> How can we stay with basics, if it really makes coding harder for everyone?

See when typing a dot in TypeScript or JavaScript not "basic snippets" are computed but the truth, which is member completion. Many people have complained that we add to that finite set "random" snippets, consider that not everyone has the majority of snippets designed to come after the dot. 

I hope you understand that there are folks that want the opposite and that we cannot change default behaviour with ease. I believe that the above settings can make you life a lot easier. Also not that there are different ways to insert a snippet. One is IntelliSense, but there is also tab-completion, the insert-snippet-action, and the ability to bind a keybinding to a snippet.

**Comment by user3323** [comment]
@jrieken, sorry for bothering you, but for me it's really huge problem.

> "javascript.useCodeSnippetsOnMethodSuggest": true

Well, there is no such setting:
![image](https://user-images.githubusercontent.com/24613274/29370609-8dcac022-82ae-11e7-8511-5108e6fb480e.png)

> "typescript.useCodeSnippetsOnMethodSuggest"

Adding just parenthesis...
Can you say where I can find all basic snippets, so I could edit them?
That will be the only thing that would help me.

> I hope you understand that there are folks that want the opposite and that we cannot change default behaviour with ease.

That's why I asked you with all my respect, if you can, to do an OPTION (in settings), that by default would be false, but if someone wants to see custom snippets right after typing the dots - turn it to true and will get them.

> the insert-snippet-action

do you mean that?
![image](https://user-images.githubusercontent.com/24613274/29370988-b71c0674-82af-11e7-98a3-20fc8c7952b1.png)

> and the ability to bind a keybinding to a snippet.

Too bad, but there are not so many keys on keyboard, so that's not gonna work.
